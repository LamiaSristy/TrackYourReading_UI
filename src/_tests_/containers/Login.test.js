import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Login from '../../containers/Login';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <Login />
  </Provider>,
);

describe('Login Container', () => {
  let container;
  beforeEach(() => {
    container = setup();
  });

  test('renders the body of the page', () => {
    const page = container.find('.body');
    expect(page).toMatchSnapshot();
  });

  test('renders the text of the login class', () => {
    const text = container.find('.login');
    expect(text).toMatchSnapshot();
  });

  test('renders the text of the btn-login class', () => {
    const text = container.find('.btn-login');
    expect(text).toMatchSnapshot();
  });

  test('renders the text of the btn-signup class', () => {
    const text = container.find('.btn-signup');
    expect(text).toMatchSnapshot();
  });

  test('renders the text of the errors-div class', () => {
    const text = container.find('.errors-div');
    expect(text).toMatchSnapshot();
  });
});
