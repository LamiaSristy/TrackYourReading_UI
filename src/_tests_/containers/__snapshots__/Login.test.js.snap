// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container renders the body of the page 1`] = `ShallowWrapper {}`;

exports[`Login Container renders the text of the btn-login class 1`] = `ShallowWrapper {}`;

exports[`Login Container renders the text of the btn-signup class 1`] = `ShallowWrapper {}`;

exports[`Login Container renders the text of the errors-div class 1`] = `ShallowWrapper {}`;

exports[`Login Container renders the text of the login class 1`] = `ShallowWrapper {}`;
